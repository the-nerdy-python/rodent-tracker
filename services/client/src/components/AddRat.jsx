import React from 'react';

const AddRat = (props) => {
  return (
    <form onSubmit={(event) => props.addRat(event)}>
      <div className="field">
        <input
          name="color"
          className="input is-large"
          type="text"
          placeholder="Enter a color"
          required
          value={props.color}
          onChange={props.handleChange}
        />
      </div>
      <div className="field">
        <input
          name="weight"
          className="input is-large"
          type="email"
          placeholder="Enter an weight in grams"
          required
          value={props.weight}
          onChange={props.handleChange}
        />
      </div>
      <input
        type="submit"
        className="button is-primary is-large is-fullwidth"
        value="Submit"
      />
    </form>
  )
};

export default AddRat;
