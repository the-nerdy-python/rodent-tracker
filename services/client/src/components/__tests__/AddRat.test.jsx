import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import AddRat from '../AddRat';

test('AddRat renders properly', () => {
  const wrapper = shallow(<AddRat/>);
  const element = wrapper.find('form');
  expect(element.find('input').length).toBe(3);
  expect(element.find('input').get(0).props.name).toBe('color');
  expect(element.find('input').get(1).props.name).toBe('weight');
  expect(element.find('input').get(2).props.type).toBe('submit');
});

test('AddRat renders a snapshot properly', () => {
  const tree = renderer.create(<AddRat/>).toJSON();
  expect(tree).toMatchSnapshot();
});

