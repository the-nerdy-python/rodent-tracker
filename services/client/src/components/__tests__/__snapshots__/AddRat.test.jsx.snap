// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRat renders a snapshot properly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="color"
      placeholder="Enter a color"
      required={true}
      type="text"
    />
  </div>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="weight"
      placeholder="Enter an weight in grams"
      required={true}
      type="email"
    />
  </div>
  <input
    className="button is-primary is-large is-fullwidth"
    type="submit"
    value="Submit"
  />
</form>
`;
